chia/wallet/wallet_node.py

```diff
@@ -178,17 +179,41 @@ def rollback_request_caches(self, reorg_height: int):
     async def get_key_for_fingerprint(self, fingerprint: Optional[int]) -> Optional[PrivateKey]:
         try:
             keychain_proxy = await self.ensure_keychain_proxy()
+            # Returns first private key if fingerprint is None
             key = await keychain_proxy.get_key_for_fingerprint(fingerprint)
         except KeyringIsEmpty:
             self.log.warning("No keys present. Create keys with the UI, or with the 'chia keys' program.")
             return None
+        except KeyringKeyNotFound:
+            self.log.warning(f"Key not found for fingerprint {fingerprint}")
+            return None
         except KeyringIsLocked:
             self.log.warning("Keyring is locked")
             return None
         except KeychainProxyConnectionFailure as e:
             tb = traceback.format_exc()
             self.log.error(f"Missing keychain_proxy: {e} {tb}")
             raise e  # Re-raise so that the caller can decide whether to continue or abort
+
+        return key
+
+    async def get_private_key(self, fingerprint: Optional[int]) -> Optional[PrivateKey]:
+        """
+        Attempt to get the private key for the given fingerprint. If the fingerprint is None,
+        get_key_for_fingerprint() will return the first private key. Similarly, if a key isn't
+        returned for the provided fingerprint, the first key will be returned.
+        """
+        key: Optional[PrivateKey] = await self.get_key_for_fingerprint(fingerprint)
+
+        if key is None:
```

Comment:
```suggestion
        if key is None and fingerprint is not None:
```

Otherwise you double try in the `fingerprint=None` case because the above `get_key_for_fingerprint(None)` already failed.